src/SuperHero.java

```diff
@@ -1,4 +1,52 @@
 public class SuperHero {
+    int healthPoints = 100;
+    int attackPower;
+    boolean isTurningIntoStatue;
+    boolean isHeroKind;
+    String name;
 
-    // Тут мають бути властивості і поведінка Супергероя
+    public void attack(SuperHero opponent) {
+        if (isTurningIntoStatue || healthPoints <= 0) {
+            return;
+        }
+        System.out.println("SuperHero " + this.name + "  атакує і завдає " + attackPower + " очок ушкоджень ");
+        opponent.takeDamage(attackPower);
+    }
+
+    public void takeDamage(int damage) {
+        /*    System.out.println("SuperHero " + this.name + "  receives " + damage); */
+        healthPoints -= damage;
+        System.out.println("SuperHero " + this.name + "  має  залишок " + healthPoints + " очок здоров'я ");
+        if (healthPoints <= 0) {
+            turningIntoStatue();
```

Comment:
I love how you did it in a humanistic way :) no dying, but turning into the statue :)